src/main/java/games/strategy/triplea/ui/EditPanel.java

```diff
@@ -537,31 +537,20 @@ private static void sortUnitsToRemove(final List<Unit> units) {
   }
 
   private static Comparator<Unit> getRemovableUnitsOrder() {
-    final Comparator<Unit> removableUnitsOrder = (unit1, unit2) -> {
-      final TripleAUnit u1 = TripleAUnit.get(unit1);
-      final TripleAUnit u2 = TripleAUnit.get(unit2);
+    return Comparator.comparing(TripleAUnit::get, (u1, u2) -> {
       if (UnitAttachment.get(u1.getType()).getTransportCapacity() != -1) {
-
         // Sort by decreasing transport capacity
-        final Collection<Unit> transporting1 = u1.getTransporting();
-        final Collection<Unit> transporting2 = u2.getTransporting();
-        final int cost1 = TransportUtils.getTransportCost(transporting1);
-        final int cost2 = TransportUtils.getTransportCost(transporting2);
-        if (cost1 != cost2) {
-          return cost2 - cost1;
-        }
+        return Comparator.<TripleAUnit, Collection<Unit>>comparing(TripleAUnit::getTransporting,
+            Comparator.comparingInt(TransportUtils::getTransportCost).reversed())
+            .thenComparingInt(TripleAUnit::getMovementLeft)
+            .thenComparingInt(Object::hashCode)
+            .compare(u1, u2);
       }
-
       // Sort by increasing movement left
```

Comment:
This comment now applies to both the preceding and following blocks.  It might be helpful to a future reader to change the previous comment to read

```java
// Sort by decreasing transport capacity and then by increasing movement left
```

However, I think the refactored code makes these comments superfluous anyway.  So, if you want to remove them outright, I'm not going to complain. :smile: